### ISSUE TITLE
Form Closing event Cancel=True not stopping Tab Closing

### ISSUE THREAD

**Issue by digao-dalpiaz** [lstratman/EasyTabs#30]
Hello, 

I'm using EasyTabs and I am not able to figure out how to stop closing a tab.

The Form has an event FormClosing, so I set `e.Cancel = True`, but even with this the form Tab is closing undesirably.

Let me say EasyTabs is incredible project! Thanks.
Regards.

**Comment by maihcx** [comment]
You can use "Overide Protected Void OnClosing" to fix it.

**Comment by maihcx** [comment]
![image](https://user-images.githubusercontent.com/59072697/79771424-f2ace800-8358-11ea-9481-11bf8b00359a.png)
i customer this for new UI.

**Comment by digao-dalpiaz** [comment]
> You can use "Overide Protected Void OnClosing" to fix it.

I didn't understand.

In TitleBarTabs.cs there is a method TitleBarTabs_Closing:

```
private void TitleBarTabs_Closing(object sender, CancelEventArgs e)
		{
			TitleBarTab tab = (TitleBarTab) sender;
			CloseTab(tab);

			if (!tab.Content.IsDisposed && AeroPeekEnabled)
			{
				TaskbarManager.Instance.TabbedThumbnail.RemoveThumbnailPreview(tab.Content);
			}

			if (_overlay != null)
			{
				_overlay.Render(true);
			}
		}
```

I think should have an If in this method to check Cancel=True and stop tab closing.

If I override closing method, this internal method still closing tab.

**Comment by digao-dalpiaz** [comment]
Now it seems to be working correctly. I added a condition in TitleBarTabs_Closing:

```csharp
private void TitleBarTabs_Closing(object sender, CancelEventArgs e)
{
	if (e.Cancel) return;
	...
}
```

**Comment by vdemichev** [comment]
Hi Luke,

I have been using EasyTabs and it's fantastic! Many thanks!
This is the only bug so far :) Would it be possible to fix it as suggested?

Best wishes,

Vadim

**Comment by lstratman** [comment]
A pull request has been merged for this and I've added some additional code to explicitly call `Close()` on each tab form when the application form's `FormClosing` event is triggered.  If any of the tabs cancel their close (`e.Cancel = true`) then the overall application's `FormClosing` event is cancelled and the application stays open.

This is in the 2.0.0-beta2 version of the package that I just published.  I'm going to integrate this with another project of mine and let it bake for a few days before publishing an "official" 2.0.0 version.  Feel free to check it out and let me know if you run into any issues.
